CI: wheel builds for Python 3.13 are failing

[Example log](https://github.com/scipy/scipy/actions/runs/10122442825/job/27994599772)

Summary:
- all the `cp313` and `cp313t` wheel builds are failing (Linux/macOS/Windows)
- they were working two days ago
- 3.13 builds in regular CI are all working fine
- nothing seems to have changed:
    - no new `cibuildwheel`,
    - I verified it's not the new `meson` 1.5.1 (and the first wheel build failures were hours before 1.5.1 was released)
    - no new `pip`, `build`, `manylinux` images
    - the jobs appear to still be using CPython 3.13.0b2, and anyway 3.13.0b4 is already 10 days old
- the failure mode seems related to `numpy` (see traceback below), _however_ if there was an actual problem with `numpy` then regular CI would be failing as well.
- PyWavelets wheel builds are failing with the same problem, and also can't find `numpy-config`. That indicates an issue with install paths (both `bin/` and `lib/python3.13[t]/site-packages` go missing apparently)

```
     ..\scipy\meson.build:48:17: ERROR: Command `C:\Users\runneradmin\AppData\Local\Temp\cibw-run-vyehvmvp\cp313-win_amd64\build\venv\Scripts\python.exe -c "import os
    os.chdir(os.path.join(\"..\", \"tools\"))
    import numpy as np
    try:
      incdir = os.path.relpath(np.get_include())
    except Exception:
      incdir = np.get_include()
    print(incdir)
      "` failed with status 1.
    
    A full log can be found at D:\a\scipy\scipy\.mesonpy-zfcnoua5\meson-logs\meson-log.txt
    [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
  error: metadata-generation-failed
```

`build/meson-logs/meson-log.txt` is hard to access when `cibuildwheel` is used, because `cibuildwheel` doesn't give you any way to run another command after a failure, or modify the install command. I don't expect it to be particularly informative though.

The best way to debug this is probably to reproduce this locally with `cibuildwheel`, and then use [`CIBW_DEBUG_KEEP_CONTAINER`](https://cibuildwheel.pypa.io/en/stable/options/#cibw_debug_keep_container) and try to figure out what is happening.

Locally I'm on macOS M1, trying to build through docker.

## cp313_manylinux_aarch64
https://cirrus-ci.com/build/5141928801468416 works on cirrus.

## cp313_manylinux_aarch64
- I could successfully build and test locally

## cp313t-manylinux_aarch64
- I had issues building locally because of:

> + python -m pip install -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple numpy cython
Looking in indexes: https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
ERROR: Could not find a version that satisfies the requirement numpy (from versions: none)
ERROR: No matching distribution found for numpy

that's because there's no numpy prerelease wheel for that. It doesn't go on to try to build numpy.

## cp313_manylinux_x86_64
- I then tried to build cp313_manylinux_x86_64 using `cibuildwheel --platform linux --archs x86_64` and `CIBW_BUILD=cp313t-manylinux*`. This one was a little weird. It downloads and installs the correct numpy at the start of the run:

> Collecting numpy
  Downloading https://pypi.anaconda.org/scientific-python-nightly-wheels/simple/numpy/2.1.0.dev0/numpy-2.1.0.dev0-cp313-cp313t-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (19.2 MB)

But then during the scipy build it tries to build numpy (*2.01*)!:

<details open>
<summary>build log</summary>
```
Preparing metadata (pyproject.toml) did not run successfully.
    exit code: 1
        [864 lines of output]
+ /opt/python/cp313-cp313t/bin/python /tmp/pip-install-yxcc5wye/numpy_47b22e5d30a643cab7c1c21298207a13/vendored-meson/meson/meson.py setup /tmp/pip-install-yxcc5wye/numpy_47b22e5d30a643cab7c1c21298207a13 /tmp/pip-install-yxcc5wye/numpy_47b22e5d30a643cab7c1c21298207a13/.mesonpy-06cjpk_v -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/tmp/pip-install-yxcc5wye/numpy_47b22e5d30a643cab7c1c21298207a13/.mesonpy-06cjpk_v/meson-python-native-file.ini

            The Meson build system
            Version: 1.4.99
            Source dir: /tmp/pip-install-yxcc5wye/numpy_47b22e5d30a643cab7c1c21298207a13
            Build dir: /tmp/pip-install-yxcc5wye/numpy_47b22e5d30a643cab7c1c21298207a13/.mesonpy-06cjpk_v
            Build type: native build
            Project name: NumPy
            Project version: 2.0.1
            C compiler for the host machine: cc (gcc 10.2.1 "cc (GCC) 10.2.1 20210130 (Red Hat 10.2.1-11)")
            C linker for the host machine: cc ld.bfd 2.35-5
            C++ compiler for the host machine: c++ (gcc 10.2.1 "c++ (GCC) 10.2.1 20210130 (Red Hat 10.2.1-11)")
            C++ linker for the host machine: c++ ld.bfd 2.35-5
            Cython compiler for the host machine: cython (cython 3.0.10)
            Host machine cpu family: x86_64
            Host machine cpu: x86_64
            Program python found: YES (/opt/python/cp313-cp313t/bin/python)

SNIP

            FAILED: numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/meson-generated_numpy_random__mt19937.pyx.c.o
            cc -Inumpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p -Inumpy/random -I../numpy/random -I../numpy/random/src -Inumpy/_core -I../numpy/_core -Inumpy/_core/include -I../numpy/_core/include -I../numpy/_core/src/common -Inumpy -I/opt/_internal/cpython-3.13.0b3-nogil/include/python3.13t -I/tmp/pip-install-yxcc5wye/numpy_47b22e5d30a643cab7c1c21298207a13/.mesonpy-06cjpk_v/meson_cpu -fvisibility=hidden -fdiagnostics-color=always -DNDEBUG -Wall -Winvalid-pch -std=c11 -O3 -fno-strict-aliasing -msse -msse2 -msse3 -DNPY_HAVE_SSE2 -DNPY_HAVE_SSE -DNPY_HAVE_SSE3 -fPIC -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -DNPY_NO_DEPRECATED_API=0 -MD -MQ numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/meson-generated_numpy_random__mt19937.pyx.c.o -MF numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/meson-generated_numpy_random__mt19937.pyx.c.o.d -o numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/meson-generated_numpy_random__mt19937.pyx.c.o -c numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c
            numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:2380:80: error: unknown type name ‘__pyx_vectorcallfunc’; did you mean ‘vectorcallfunc’?
             2380 | static CYTHON_INLINE PyObject *__Pyx_PyVectorcall_FastCallDict(PyObject *func, __pyx_vectorcallfunc vc, PyObject *const *args, size_t nargs, PyObject *kw);
                  |                                                                                ^~~~~~~~~~~~~~~~~~~~
                  |                                                                                vectorcallfunc
            numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11238:69: error: unknown type name ‘__pyx_vectorcallfunc’; did you mean ‘vectorcallfunc’?
            11238 | static PyObject *__Pyx_PyVectorcall_FastCallDict_kw(PyObject *func, __pyx_vectorcallfunc vc, PyObject *const *args, size_t nargs, PyObject *kw)
                  |                                                                     ^~~~~~~~~~~~~~~~~~~~
                  |                                                                     vectorcallfunc
            numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11283:80: error: unknown type name ‘__pyx_vectorcallfunc’; did you mean ‘vectorcallfunc’?
            11283 | static CYTHON_INLINE PyObject *__Pyx_PyVectorcall_FastCallDict(PyObject *func, __pyx_vectorcallfunc vc, PyObject *const *args, size_t nargs, PyObject *kw)
                  |                                                                                ^~~~~~~~~~~~~~~~~~~~
                  |                                                                                vectorcallfunc
            numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c: In function ‘__Pyx_CyFunction_CallAsMethod’:
            numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11972:6: error: unknown type name ‘__pyx_vectorcallfunc’; did you mean ‘vectorcallfunc’?
            11972 |      __pyx_vectorcallfunc vc = __Pyx_CyFunction_func_vectorcall(cyfunc);
                  |      ^~~~~~~~~~~~~~~~~~~~
                  |      vectorcallfunc
            numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:2470:45: warning: initialization of ‘int’ from ‘vectorcallfunc’ {aka ‘struct _object * (*)(struct _object *, struct _object * const*, long unsigned int,  struct _object *)’} makes integer from pointer without a cast [-Wint-conversion]
             2470 | #define __Pyx_CyFunction_func_vectorcall(f) (((PyCFunctionObject*)f)->vectorcall)
                  |                                             ^
            numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11972:32: note: in expansion of macro ‘__Pyx_CyFunction_func_vectorcall’
            11972 |      __pyx_vectorcallfunc vc = __Pyx_CyFunction_func_vectorcall(cyfunc);
                  |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11975:16: warning: implicit declaration of function ‘__Pyx_PyVectorcall_FastCallDict’; did you mean ‘__Pyx_PyObject_FastCallDict’? [-Wimplicit-function-declaration]
            11975 |         return __Pyx_PyVectorcall_FastCallDict(func, vc, &PyTuple_GET_ITEM(args, 0), (size_t)PyTuple_GET_SIZE(args), kw);
                  |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                  |                __Pyx_PyObject_FastCallDict
            numpy/random/_mt19937.cpython-313t-x86_64-linux-gnu.so.p/numpy/random/_mt19937.pyx.c:11975:16: warning: returning ‘int’ from a function with return type ‘PyObject *’ {aka ‘struct _object *’} makes pointer from integer without a cast [-Wint-conversion]
            11975 |         return __Pyx_PyVectorcall_FastCallDict(func, vc, &PyTuple_GET_ITEM(args, 0), (size_t)PyTuple_GET_SIZE(args), kw);
```
</details>

--- a/.github/workflows/free_threaded_wheels.yml
+++ b/.github/workflows/free_threaded_wheels.yml
@@ -158,7 +158,7 @@ jobs:
           echo "CIBW_REPAIR_WHEEL_COMMAND_MACOS=$CIBW" >> "$GITHUB_ENV"
 
       - name: Build wheels
-        uses: pypa/cibuildwheel@a8d190a111314a07eb5116036c4b3fb26a4e3162  # v2.19.0
+        uses: pypa/cibuildwheel@7e5a838a63ac8128d71ab2dfd99e4634dd1bca09  # v2.19.2
         env:
           CIBW_PRERELEASE_PYTHONS: True
           CIBW_FREE_THREADED_SUPPORT: True

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -37,7 +37,7 @@ jobs:
   get_commit_message:
     name: Get commit message
     runs-on: ubuntu-latest
-    if: github.repository == 'scipy/scipy'
+    if: github.repository == 'andyfaff/scipy'
     outputs:
       message: ${{ steps.commit_message.outputs.message }}
     steps:
@@ -207,7 +207,7 @@ jobs:
           echo "CIBW_BUILD_FRONTEND=$CIBW" >> "$GITHUB_ENV" 
 
       - name: Build wheels
-        uses: pypa/cibuildwheel@v2.19.1
+        uses: pypa/cibuildwheel@v2.19.2
         env:
           CIBW_BUILD: ${{ matrix.python[0] }}-${{ matrix.buildplat[1] }}*
           CIBW_ARCHS: ${{ matrix.buildplat[2] }}

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -37,7 +37,7 @@ jobs:
   get_commit_message:
     name: Get commit message
     runs-on: ubuntu-latest
-    if: github.repository == 'andyfaff/scipy'
+    if: github.repository == 'scipy/scipy'
     outputs:
       message: ${{ steps.commit_message.outputs.message }}
     steps:
@@ -79,13 +79,13 @@ jobs:
         # should also be able to do multi-archs on a single entry, e.g.
         # [windows-2019, win*, "AMD64 x86"]. However, those two require a different compiler setup
         # so easier to separate out here.
-#        - [ubuntu-22.04, manylinux, x86_64, "", ""]
-#        - [ubuntu-22.04, musllinux, x86_64, "", ""]
+        - [ubuntu-22.04, manylinux, x86_64, "", ""]
+        - [ubuntu-22.04, musllinux, x86_64, "", ""]
         - [macos-12, macosx, x86_64, openblas, "10.13"]
         - [macos-13, macosx, x86_64, accelerate, "14.0"]
         - [macos-14, macosx, arm64, openblas, "12.3"]
         - [macos-14, macosx, arm64, accelerate, "14.0"]
-#        - [windows-2019, win, AMD64, "", ""]
+        - [windows-2019, win, AMD64, "", ""]
         python: [["cp310", "3.10"], ["cp311", "3.11"], ["cp312", "3.12"], ["cp313", "3.13"]]
         # python[0] is used to specify the python versions made by cibuildwheel
 

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -79,13 +79,13 @@ jobs:
         # should also be able to do multi-archs on a single entry, e.g.
         # [windows-2019, win*, "AMD64 x86"]. However, those two require a different compiler setup
         # so easier to separate out here.
-        - [ubuntu-22.04, manylinux, x86_64, "", ""]
-        - [ubuntu-22.04, musllinux, x86_64, "", ""]
+#        - [ubuntu-22.04, manylinux, x86_64, "", ""]
+#        - [ubuntu-22.04, musllinux, x86_64, "", ""]
         - [macos-12, macosx, x86_64, openblas, "10.13"]
         - [macos-13, macosx, x86_64, accelerate, "14.0"]
         - [macos-14, macosx, arm64, openblas, "12.3"]
         - [macos-14, macosx, arm64, accelerate, "14.0"]
-        - [windows-2019, win, AMD64, "", ""]
+#        - [windows-2019, win, AMD64, "", ""]
         python: [["cp310", "3.10"], ["cp311", "3.11"], ["cp312", "3.12"], ["cp313", "3.13"]]
         # python[0] is used to specify the python versions made by cibuildwheel
 

